python/src/rejot_contract/public_schema.py

```diff
@@ -0,0 +1,52 @@
+from typing import List, Literal, Type, TypeVar, Any
+from pydantic import BaseModel
+
+class InvalidPublicSchemaError(Exception):
+    def __init__(self, message: str):
+        super().__init__(message)
+        self.name = "InvalidPublicSchemaError"
+
+class PostgresPublicSchemaConfigTransformation(BaseModel):
+    operation: Literal["insert", "update", "delete"]
+    table: str
+    sql: str
+
+class Version(BaseModel):
+    major: int
+    minor: int
+
+class PublicSchemaConfig(BaseModel):
+    publicSchemaType: Literal["postgres"]
+    transformations: List[PostgresPublicSchemaConfigTransformation]
+
+class Source(BaseModel):
+    dataStoreSlug: str
+
+class PublicSchemaData(BaseModel):
+    name: str
+    source: Source
+    outputSchema: dict
+    version: Version
+    config: PublicSchemaConfig
+
+T = TypeVar('T', bound=BaseModel)
+
+def create_public_schema(
+    public_schema_name: str,
+    source: Source,
+    output_schema: Type[T],
+    version: Version,
+    config: PublicSchemaConfig
+) -> dict[str, Any]:
```

Comment:
Wouldn't it be better if this exposed an explicit type?

Reply:
I don't think so, as it is just a raw JSON type. We would need to add a bunch of additional typing. I haven't found a function like `z.infer` for python yet.